qiskit_braket_provider/providers/adapter.py

```diff
@@ -407,10 +332,23 @@ def convert_continuous_qubit_indices(
     return target
 
 
-def convert_qiskit_to_braket_circuit(circuit: QuantumCircuit) -> Circuit:
+@singledispatch
```

Comment:
Why use singledispatch here? Do you have ideas for additional types in this qiskit-braket repo for which `to_braket` would be implemented?